src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Security/CertificateAuthenticationWithPki/CertificateAuthenticationWithPKI.xaml.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright 2019 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an 
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific 
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Portal;
+using Esri.ArcGISRuntime.Security;
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Windows.UI.Popups;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace ArcGISRuntime.UWP.Samples.CertificateAuthenticationWithPKI
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Certificate authentication with PKI",
+        "Security",
+        "Access secured portals using a certificate.",
+        "")]
+    public partial class CertificateAuthenticationWithPKI
+    {
+        public CertificateAuthenticationWithPKI()
+        {
+            InitializeComponent();
+        }
+
+        public async Task<Credential> CreateCertCredentialAsync(CredentialRequestInfo info)
```

Comment:
Did you consider setting this as the challenge handler for `AuthenticationManager`? It works fine this way (probably best for simplicity).